src/Components/Components/src/Routing/RouteTableFactory.cs

```diff
@@ -73,7 +73,9 @@ static void GetRouteableComponents(List<Type> routeableComponents, Assembly asse
         {
             foreach (var type in assembly.ExportedTypes)
             {
-                if (typeof(IComponent).IsAssignableFrom(type) && type.IsDefined(typeof(RouteAttribute)))
+                if (typeof(IComponent).IsAssignableFrom(type)
+                    && type.IsDefined(typeof(RouteAttribute))
+                    && type.GetCustomAttribute<AllowInteractiveRoutingAttribute>() is not { Allow: false })
```

Comment:
This could end up giving different results than the `HttpContext.AllowsInteractiveRouting()` extension method if someone does something like `MapRazorComponents<App>()...WithMetadata(new AllowInteractiveRoutingAttribute(false))`. This could lead people to think things are working at first only to realize it's broken later when they attempt an interactive navigation. Is there a way we can look at the final metadata from the potentially routable component's `Endpoint`?

I know this is a problem for other metadata like `[AllowAnonymous]`. Blazor not being able to "see" all the endpoint metadata [can be helpful](https://github.com/dotnet/aspnetcore/issues/53732#issuecomment-2021878231) for working around the issue many people are facing in .NET 8 where the authorization middleware rejects requests before Blazor's `AuthorizeRouteView` can handle it. That's because the authorization middleware will "see" the `[AllowAnonymous]` metadata added by `MapRazorComponents<App>()...AllowAnonymous()` and let the request through while `AuthorizeRouteView` cannot. However, I think we should fix that. And I certainly don't think we should be adding new attributes that can be interpreted one way by Blazor's core components and another code checking the `Endpoint`.

Reply:
> Is there a way we can look at the final metadata from the potentially routable component's Endpoint?

No, this code runs client-side (e.g., on WebAssembly), and can't see the server-side endpoint table.

If we later create some bigger mechanism for supplying extra info from the server-side endpoint table to the client, then we could use that here. But a new mechanism like that would be well out of scope for this issue. Until something like that exists, it's up to the developer not to cause themselves any extra problems here by thinking they can change the server-side metadata and the client would somehow know about that.

Reply:
If we wanted to absolutely ensure consistency in this case, the only reasonable option I can think of is for the server not to read the attribute from endpoint metadata at all, but instead to read it from the component type directly, so you can't customize it using `WithMetadata`.

Reply:
> If we wanted to absolutely ensure consistency in this case, the only reasonable option I can think of is for the server not to read the attribute from endpoint metadata at all, but instead to read it from the component type directly, so you can't customize it using `WithMetadata`.

I think that's how it should work for now. I like the cleanliness of just reading the metadata directly on the server side, but not if it's inconsistent with what the route table sees. If we figure out how to easily read the final metadata in both places later, we can make that change then.

Reply:
Fine with me. It's how I implemented it originally 😄